config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<!--
+  ~ SPDX-License-Identifier: Apache-2.0
+
+  ~ The OpenSearch Contributors require contributions made to
+  ~ this file be licensed under the Apache-2.0 license or a
+  ~ compatible open source license.
+  -->
+
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<!--
+    Formatting checks are done by spotless. This config is only for checks spotless doesn't do.
+-->
+
+<module name="Checker">
+    <property name="charset" value="UTF-8" />
+
+    <module name="SuppressionFilter">
+        <property name="file" value="${config_loc}/checkstyle_suppressions.xml" />
+    </module>
+
+    <module name="TreeWalker">
+        <!-- Disallows star imports -->
+        <module name="AvoidStarImport" />
+        
+        <!--  Requires javadoc on public and protected interfaces, classes, enums -->
+        <module name="MissingJavadocType">
+            <property name="scope" value="protected"/>
+        </module>
+        
+        <!-- Requires javadoc on public and protected methods, excluding getters and setters -->
+        <module name="MissingJavadocMethod">
+            <property name="scope" value="protected"/>
+            <property name="allowMissingPropertyJavadoc" value="true"/>
+        </module>
+        
+        <!--  Checks that javadocs contain correct param and return -->
+        <module name="JavadocMethod">
+            <property name="id" value="JavadocMethod"/>
+            <property name="accessModifiers" value="public, protected"/>
+        </module>
+
+        <!-- Allows missing returns on extension interface methods -->
+        <module name="JavadocMethod">
+            <property name="id" value="JavadocMethodAllowMissingReturnTag"/>
+            <property name="accessModifiers" value="public, protected"/>
+            <property name="allowMissingReturnTag" value="true"/>
+        </module>
+        
```

Comment:
We can also add ` <module name="NewlineAtEndOfFile"/>` and remove Linelint. One less of a tool :)
Also, in the future we can have more strict checks for VariableName, MethodName and ParameterName

Reply:
> We already have removed LineLint as spotless handles end of lines. (Should do the same on OpenSearch!)

@dbwiddis https://github.com/opensearch-project/OpenSearch/pull/6657